src/zarr/core/group.py

```diff
@@ -697,6 +697,73 @@ def from_dict(
             store_path=store_path,
         )
 
+    async def copy_store(
```

Comment:
we should think carefully about the name -- technically this method doesn't copy the store, it only copies the arrays / groups contained under this group, so I'm not sure "copy_store" is the best description. Plain "copy" is IMO overloaded with making an in-memory copy of the group object. Alternatives: "copy_to", "resave", @zarr-developers/python-core-devs any other ideas?

Reply:
I like copy_to.